### PR TITLE
[v1.3][W14-B2] PTMan

### PR DESCRIPTION
**Pull request by shanwpf**
@koo1993 @sunbangjie @hzxcaryn

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -1,11 +1,9 @@
 = Address Book (Level 4)
 ifdef::env-github,env-browser[:relfileprefix: docs/]
 
-https://travis-ci.org/se-edu/addressbook-level4[image:https://travis-ci.org/se-edu/addressbook-level4.svg?branch=master[Build Status]]
-https://ci.appveyor.com/project/damithc/addressbook-level4[image:https://ci.appveyor.com/api/projects/status/3boko2x2vr5cc3w2?svg=true[Build status]]
-https://coveralls.io/github/se-edu/addressbook-level4?branch=master[image:https://coveralls.io/repos/github/se-edu/addressbook-level4/badge.svg?branch=master[Coverage Status]]
-https://www.codacy.com/app/damith/addressbook-level4?utm_source=github.com&utm_medium=referral&utm_content=se-edu/addressbook-level4&utm_campaign=Badge_Grade[image:https://api.codacy.com/project/badge/Grade/fc0b7775cf7f4fdeaf08776f3d8e364a[Codacy Badge]]
-https://gitter.im/se-edu/Lobby[image:https://badges.gitter.im/se-edu/Lobby.svg[Gitter chat]]
+https://travis-ci.org/CS2103JAN2018-W14-B2/main[image:https://travis-ci.org/CS2103JAN2018-W14-B2/main.svg?branch=master[Build Status]]
```
Comment (okkhoy) [inline README.adoc:4]: Please update the title of your project! It is not AddressBook anymore.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -798,35 +798,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
-=== Use case: Delete person
+=== Use case: Add employee
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  Employer enters employee details and working outlet(s)
```
Comment (okkhoy) [inline docs/DeveloperGuide.adoc]: Maybe you can reword this `Employer enters employee details and working outlets to add employee to the system`

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -840,18 +904,124 @@ Use case ends.
...
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `1.8.0_60` or higher installed.
+.  Should work on both 32-bit and 64-bit environment.
 .  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Should respond within 2 seconds after entering any command.
+.  Should be usable by users who has never used a command line interface before.
+.  Should be expected to manage contact details of part-time workers in any outlet.
```
Comment (okkhoy) [inline docs/DeveloperGuide.adoc]: should be `able` to...

File: src/main/java/seedu/address/model/AddressBook.java [hunk 4]
```diff
@@ -155,6 +156,58 @@ public void addTag(Tag t) throws UniqueTagList.DuplicateTagException {
 
     //// util methods
 
+    /**
+     * Remove tag from Person if the tag exist in Person.
+     * @param tag
+     * @param person
+     */
+    private void removeTagFromPerson(Tag tag, Person person) {
```
Comment (okkhoy) [inline src/main/java/seedu/address/model/AddressBook.java]: There is no update to the user guide for this feature. Remember, any new feature or change in functionality should ALWAYS be accompanied by an update to the tests and documentation.

File: src/main/java/seedu/address/storage/XmlAddressBookStorage.java [hunk 5]
```diff
@@ -79,4 +80,22 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) th
         XmlFileStorage.saveDataToFile(file, new XmlSerializableAddressBook(addressBook));
     }
 
+    @Override
+    public void backupAddressBook(ReadOnlyAddressBook addressBook) throws IOException {
```
Comment (okkhoy) [inline src/main/java/seedu/address/storage/XmlAddressBookStorage.java]: Same comment as above!

**Review by okkhoy — COMMENTED** [review]
I have a feeling that the scope of the user stories recorded is quite big. You may want to re-prioritize on the deliverables.
Good job on this!

**Comment by okkhoy** [comment]
@shanwpf @koo1993 @SunBangjie @hzxcaryn

File: README.adoc [hunk 6]
```diff
@@ -15,25 +13,27 @@ ifndef::env-github[]
...
-** More test cases, including automated GUI testing.
-** Support for _Build Automation_ using Gradle and for _Continuous Integration_ using Travis CI.
+* This is a desktop application for both part-time employers and employees. It has a GUI but most of the user interactions happen using a CLI (Command Line Interface).
+* For employers:
+. Provides easy management of employees.
+. Indicate time slots where manpower is required.
+* For employees:
+. View the available working time slots through the timetable.
+. Register for preferred time slots that you want to work at.
+* A sophisticated GUI that includes a list panel and an in-built timetable.
```
Comment (okkhoy) [inline README.adoc:23]: Just a question, do you manage allotment of employees to time slots as well?

File: README.adoc [hunk 7]
```diff
@@ -15,25 +13,27 @@ ifndef::env-github[]
...
 * <<UserGuide#, User Guide>>
 * <<DeveloperGuide#, Developer Guide>>
-* <<LearningOutcomes#, Learning Outcomes>>
 * <<AboutUs#, About Us>>
 * <<ContactUs#, Contact Us>>
 
 == Acknowledgements
 
-* Some parts of this sample application were inspired by the excellent http://code.makery.ch/library/javafx-8-tutorial/[Java FX tutorial] by
+* Some parts of this application were inspired by the excellent http://code.makery.ch/library/javafx-8-tutorial/[Java FX tutorial] by
```
Comment (okkhoy) [inline README.adoc:36]: Need to attribute SE-EDU addressbook-level4 as well

File: docs/AboutUs.adoc [hunk 8]
```diff
@@ -3,53 +3,53 @@
...
-=== Benson Meier
-image::yl_coder.jpg[width="150", align="left"]
-{empty}[http://github.com/yl-coder[github]] [<<johndoe#, portfolio>>]
+=== Sun Bangjie
+image::SunBangjie.JPG[width="150", align="left"]
+{empty}[http://github.com/SunBangjie[github]] [<<SunBangjie#, portfolio>>]
 
 Role: Developer +
-Responsibilities: UI
+Responsibilities: Integration + Model component + Documentation
```
Comment (okkhoy) [inline docs/AboutUs.adoc]: Can you indicate one of you as the team lead?

File: docs/UserGuide.adoc [hunk 9]
```diff
@@ -11,226 +12,341 @@ ifdef::env-github[]
...
+=== Employer Features
+These commands are privileged, and can only be executed by the employer. These commands requires the employer's admin password.
+
+[CAUTION]
+Admin password must always be entered as the *last* parameter of a command.
+
+==== Adding an employee: `add`
+
+Adds an employee to PTMan. +
+*Format:* `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS s/SALARY [t/TAG]... pw/ADMIN_PASSWORD` +
```
Comment (okkhoy) [inline docs/UserGuide.adoc]: 1. Do you mean that for every employee to be added, the employer has to enter his password? If so, think how you can make employers' life simple

2. Are you asking for the password in plain text? :confused:

File: docs/UserGuide.adoc [hunk 10]
```diff
@@ -11,226 +12,341 @@ ifdef::env-github[]
...
 
-=== Editing a person : `edit`
+Edits an existing employee in PTMan. +
+*Format:* `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [s/SALARY] [t/TAG]... pw/ADMIN_PASSWORD` +
+*Shorthand:* `e INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [s/SALARY] [t/TAG]... pw/ADMIN_PASSWORD`
 
-Edits an existing person in the address book. +
-Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]...`
+[NOTE]
+An employer password can't be edited. [refer to `rp` (reset password)]
```
Comment (okkhoy) [inline docs/UserGuide.adoc]: I can see `rp` only resets employee password. What about the employer? Can't be edited is acceptable, but can't be reset is a major gap!

File: docs/UserGuide.adoc [hunk 11]
```diff
@@ -11,226 +12,341 @@ ifdef::env-github[]
...
+* `list` +
+`undo pw/ADMIN_PASSWORD` +
 The `undo` command fails as there are no undoable commands executed previously.
 
-* `delete 1` +
-`clear` +
-`undo` (reverses the `clear` command) +
-`undo` (reverses the `delete 1` command) +
+* `delete 1 pw/ADMIN_PASSWORD` +
+`clear pw/ADMIN_PASSWORD` +
```
Comment (okkhoy) [inline docs/UserGuide.adoc]: I think asking for the password to be entered everytime is a hassle + security issue. You may have someone peeping into your screen! :stuck_out_tongue_winking_eye:

**Review by okkhoy — COMMENTED** [review]
Good job guys! I see this as a comprehensive update!

May I suggest adding something to be done in the future (v2.0)?

**Comment by okkhoy** [comment]
@shanwpf @SunBangjie @hzxcaryn @koo1993 

Please keep up the good work!

I would want you to mention the following in your **Developer guide**
1. For each team member what are the features (major and minor) you are proposing?
2. Within the scope of the project, how does it fit in (just a one or 2 line summary)
Please do this by end of day Monday 19 March.

File: README.adoc (hunk 7 above)
Comment (waynekoo) [reply README.adoc:36]: thanks for the input we updated the readme

File: docs/AboutUs.adoc (hunk 8 above)
Comment (waynekoo) [reply docs/AboutUs.adoc]: we have indicated the team lead :D

File: docs/UserGuide.adoc (hunk 10 above)
Comment (waynekoo) [reply docs/UserGuide.adoc]: we rethought through and came out with a solution that allows both employee and employer to reset their password. I will be reflected in user guide.

File: docs/DeveloperGuide.adoc [hunk 12]
```diff
@@ -230,17 +240,116 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+Example: `new Capacity("4")` creates a `Capacity` object representing integer `4`.
+
+In addition to these classes, `Shift` also contains a `UniqueEmployeeList` which is empty when `Shift` is constructed.
+This is used to store employees that have applied for this shift.
+
+To store a list of shifts, we may use a `UniqueShiftList` to ensure there are no duplicate shifts.
+
+The following commands directly interact with the `Shift` class:
+
+. `AddShiftCommand`: Creates a `Shift` and adds it to the `UniqueShiftList` in `OutletInformation`.
```
Comment (okkhoy) [inline docs/DeveloperGuide.adoc]: Consider using a class diagram to depict these details

File: docs/DeveloperGuide.adoc [hunk 13]
```diff
@@ -230,17 +240,116 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+The `TimetablePanel` is initialised in the `MainWindow` class in the following way:
+----
+TimetablePanel timetablePanel = new TimetablePanel(logic.getFilteredShiftList());
+----
+
+`TimetablePanel` takes in an `ObservableList<Shift>` object, which is an unmodifiable sorted view of the filtered shifts.
+ `TimetablePanel` then uses these `Shift` objects to set the Entries in the timetable view, in the method `setShifts()`.+
+
+The timetable GUI automatically refreshes to update the timetable view by calling the `updateTimetableView()` method whenever
```
Comment (okkhoy) [inline docs/DeveloperGuide.adoc]: How does auto-refresh work? You can show that in a SD.

File: docs/DeveloperGuide.adoc [hunk 14]
```diff
@@ -334,30 +458,141 @@ image::UndoRedoActivityDiagram.png[width="650"]
...
+
+===== Model component
+
+Currently, the `OutletInformation` object initialized in the `PartTimeManager` is created by calling its default constructor:
+----
+OutletInformation()
+----
+This default constructor initializes the five fields of `OutletInformation` to respective default values:
+----
+    public OutletInformation() {
```
Comment (okkhoy) [inline docs/DeveloperGuide.adoc]: Please use code snippets minimally.
Reason: it is difficult to maintain these in the document, particularly if the implementation changes.

File: docs/DeveloperGuide.adoc [hunk 15]
```diff
@@ -334,30 +458,141 @@ image::UndoRedoActivityDiagram.png[width="650"]
...
+        this.timetable = new Timetable(LocalDate.now());
+    }
+----
+This saves the time of any employer new to PTMan because the creation of an outlet has already been done with default values.
+Employers can edit those values by entering an `editoutlet` command.
+However, there is an alternative way to create an `OutletInformation` object:
+----
+OutletInformation(OutletName outletName, OperatingHours operatingHours, OutletContact outletContact, Timetable timetable)
+----
+This constructor allows flexity in storing outlet information, but master password must be set to the default value.
```
Comment (okkhoy) [inline docs/DeveloperGuide.adoc]: flexity? did you mean flexibility?

**Review by okkhoy — COMMENTED** [review]
Some of the features' implmentations are well described while others are too brief. Please update such feature implementation details.

Please use various UML diagrams to explain your implementation.

**Comment by okkhoy** [comment]
@shanwpf @koo1993 @SunBangjie @hzxcaryn 
Some comments added to the DG. Please take a look and update as necessary.

**Pull request closed without merging**